### PR TITLE
Fix/aws provision cluster

### LINKED ISSUE #1300: Can't provision with testfest config
http://uberjenkins.sc.couchbase.com/view/Utils/job/aws-provision-cluster/56/console

### PR DESCRIPTION
**Pull request by raghusarangapani**
#### Fixes #1300 .

- [x] Ran `flake8`
- [x] Ran `run_repo_tests.sh`

#### Changes proposed in this pull request:
- Ignore sync function in config.

### Successful run:
http://uberjenkins.sc.couchbase.com/view/Utils/job/aws-provision-cluster/60/console

### CODE REVIEW

**Comment by sethrosetter** [comment]
So this fix is just to get bucket names, right? This modified config is never deploy. Is that correct?

**Comment by sethrosetter** [comment]
Can you add a unit test?

**Comment by raghusarangapani** [comment]
This is used to get mode and bucket names. The actual config deployments happens through `def install_sync_gateway(cluster_config, sync_gateway_config):` in `install_sync_gateway.py` through `install-sync-gateway-package.yml`.

**Comment by raghusarangapani** [comment]
This code is in the constructor of `class Config` and not a method. I am not sure how you would unit test a constructor. Just initialize the instance and check if it doesn't blow up?

**Comment by sethrosetter** [comment]
Ok
> This is used to get mode and bucket names. The actual config deployments happens through def install_sync_gateway(cluster_config, sync_gateway_config): in install_sync_gateway.py through install-sync-gateway-package.yml.

You could do that for for unit test. If it blows up before and not any longer, then we are good.

**Comment by tleyden** [comment]
>  I am not sure how you would unit test a constructor. Just initialize the instance and check if it doesn't blow up?

Or it might be a good reason to move that functionality into a method that can be unit tested.

**Comment by raghusarangapani** [comment]
The unit test is setting some self. class variables. I can check and assert on that. Will add a unit test and update the PR.

**Comment by raghusarangapani** [comment]
Added a unit test. Verified that it fails without this fix and now it passes.

File: mobile_testkit_tests/test_config_init.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    ("sync_gateway_openid_connect_di.json", ['data-bucket', 'index-bucket'], "di"),
+    ("sync_gateway_openid_connect_unsigned_cc.json", ['data-bucket'], "cc"),
+    ("sync_gateway_openid_connect_unsigned_di.json", ['data-bucket', 'index-bucket'], "di"),
+    ("sync_gateway_sg_replicate_cc.json", ['data-bucket-1', 'data-bucket-2'], "cc"),
+    ("sync_gateway_sg_replicate_continuous_cc.json", ['data-bucket-1', 'data-bucket-2'], "cc"),
+    ("sync_gateway_sg_replicate_continuous_di.json", ['index-bucket-2', 'data-bucket-1', 'index-bucket-1', 'data-bucket', 'data-bucket-2'], "di"),
+    ("sync_gateway_sg_replicate_di.json", ['index-bucket-2', 'data-bucket-1', 'index-bucket-1', 'data-bucket', 'data-bucket-2'], "di"),
+    ("webhooks/webhook_offline_cc.json", ['data-bucket'], "cc"),
+    ("webhooks/webhook_offline_di.json", ['data-bucket', 'index-bucket'], "di"),
+    ("testfest_todo_di.json", ['data-bucket', 'index-bucket'], "di"),
```
Comment (sethrosetter) [inline mobile_testkit_tests/test_config_init.py]: Can you add todo_cc?

Comment (raghusarangapani) [inline mobile_testkit_tests/test_config_init.py]: Done, copy paste error. testfest_todo_di was in there twice

**Pull request merged**